packages/block-directory/src/components/downloadable-blocks-list/index.js

```diff
@@ -35,12 +36,19 @@ function DownloadableBlocksList( { items, onHover = noop, onSelect } ) {
 					<DownloadableBlockListItem
 						key={ item.id }
 						onClick={ () => {
-							installBlockType( item ).then( ( success ) => {
-								if ( success ) {
-									onSelect( item );
-									setIsInserterOpened( false );
-								}
-							} );
+							// Check if the block is registered (`getBlockType`
+							// will return an object). If so, insert the block.
+							// This prevents installing existing plugins.
+							if ( getBlockType( item.name ) ) {
+								onSelect( item );
+							} else {
+								installBlockType( item ).then( ( success ) => {
+									if ( success ) {
+										onSelect( item );
+										setIsInserterOpened( false );
```

Comment:
Noting that it changes the state of the inserter here but not when it's already installed. Should it align?